packages/uni_app/lib/model/entities/library_occupation.dart

```diff
@@ -18,7 +18,10 @@ class LibraryOccupation {
   late List<FloorOccupation> floors;
 
   void addFloor(FloorOccupation floor) {
-    floors.add(floor);
+    floors
+      ..add(floor)
+      ..sort((a, b) => a.number.compareTo(b.number));
```

Comment:
humn, I see what are you doing. This way you are sorting the list every time you add a new floor.
What about sorting it on rendering or find some other type in order to ensure the correct order?

Reply:
I've implemented a change to use SplayTreeSet for floor storage to achieve automatic ordering. However, this required creating a custom _floorsFromJson function since SplayTreeSet isn't natively supported by the json_annotation package. Which aproach should i take, this is okay or should i use the list and sort it instead?